app/services/TwoStepVerification.scala

```diff
@@ -38,18 +38,24 @@ trait TwoStepVerification {
 
   def twoStepVerificationThrottle: TwoStepVerificationThrottle
 
+  def upliftLocationsConfiguration: Option[String]
+
+  lazy val upliftLocations = upliftLocationsConfiguration.map(s => s.split(",").map(_.trim).map(Locations.locationFromConf).toSet).getOrElse(Set.empty)
```

Comment:
Locations.locationFromConf call has direct access to the configuration, while upliftLocationsConfiguration is injected so it is decoupled from the configuration. 
I don't know what would be a better solution but this seems to be a bit of a wrong combination to me. 
Does trait TwoStepVerification need to depend on a reference to the Play configuration or not?